### PR TITLE
10.19.0 Changelog

### PR DESCRIPTION
**Pull request by seaona**
## Explanation
Changelog for 10.19.0

## More Information
See RC Spreadhseet.

## Screenshots/Screencaps

n/a

## Manual Testing Steps
n/a

## Pre-Merge Checklist

- [X] PR template is filled out
- [ ] **IF** this PR fixes a bug, a test that _would have_ caught the bug has been added
- [ ] PR is linked to the appropriate GitHub issue
- [X] PR has been added to the appropriate release Milestone

### + If there are functional changes:

- [ ] Manual testing complete & passed
- [ ] "Extension QA Board" label has been applied

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,43 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Swaps: Standardize the spelling of 'cancelled' ([#15266](https://github.com/MetaMask/metamask-extension/pull/15266))
+- Swaps: Performance improvements for Swaps, especially on the Build Quote page (~7.2x faster) ([#15359](https://github.com/MetaMask/metamask-extension/pull/15359))
+- [FLASK] Disable "Mark all as read button" when there are no notifications ([#15333](https://github.com/MetaMask/metamask-extension/pull/15333))
+- [FLASK] Update Snaps packages to version `^0.18.1` ([#15230](https://github.com/MetaMask/metamask-extension/pull/15230))
+  - Updates the following packages from `0.16.0` to `0.18.1`:
+    - `@metamask/execution-environments`
+    - `@metamask/rpc-methods`
+    - `@metamask/snap-controllers`
+    - `@metamask/snap-types`
+  - Serialization issues were fixed in `@metmask/execution-environments`. See the ([releases](https://github.com/MetaMask/snaps-skunkworks/releases)) of the Snaps monorepo for details.
```
Comment (BeroBurny) [inline CHANGELOG.md:36]: just want to confirm `package.json` contains some of these packages on version `0.20.0` 
https://github.com/MetaMask/metamask-extension/blob/bc0d32de56995945ebae873b038cbc733ff0ad96/package.json#L136

`@metamask/rpc-methods": "^0.20.0`
`@metamask/snap-controllers": "^0.20.0`
`@metamask/snap-utils": "^0.20.0`

this release will contain them on version `0.20.0` or will be `0.18.1` ?

Comment (FrederikBolding) [reply CHANGELOG.md:51]: @BeroBurny This release will contain `0.20.0`. I have yet to update the changelog 😄

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,232 +6,71 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
-- Sync `master` with `develop` ([#15476](https://github.com/MetaMask/metamask-extension/pull/15476))
-- Fixing changelog
-- Fix
-- Merge remote-tracking branch 'origin/develop' into master-sync
-- Refactor build script to include build target ([#15447](https://github.com/MetaMask/metamask-extension/pull/15447))
-- Updating Zendesk ticket request URLs ([#15458](https://github.com/MetaMask/metamask-extension/pull/15458))
-- Update media query instances ([#15068](https://github.com/MetaMask/metamask-extension/pull/15068))
-- Convert unit test render from enzyme to @testing-library/react. ([#15091](https://github.com/MetaMask/metamask-extension/pull/15091))
-- add safegaurd for missing account in action ([#15450](https://github.com/MetaMask/metamask-extension/pull/15450))
 - [FLASK] Add update snap UI ([#15143](https://github.com/MetaMask/metamask-extension/pull/15143))
```
Comment (FrederikBolding) [inline CHANGELOG.md:72]:
```suggestion
- [FLASK] Add DApp triggered Snap updates ([#15143](https://github.com/MetaMask/metamask-extension/pull/15143))
```
Can be categorized as "added"

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,232 +6,71 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
-- Fix "app-init" injection ([#15320](https://github.com/MetaMask/metamask-extension/pull/15320))
-- Add Sign-In with Ethereum ([#14438](https://github.com/MetaMask/metamask-extension/pull/14438))
-- Moving incorrectly ordered page component to correct pages category ([#15426](https://github.com/MetaMask/metamask-extension/pull/15426))
-- remove unnecessary usage of renderableGasButton logic ([#15422](https://github.com/MetaMask/metamask-extension/pull/15422))
-- Add QA migration guide ([#14544](https://github.com/MetaMask/metamask-extension/pull/14544))
-- Add data about dropped transaction in FINALIZED metrics event ([#15383](https://github.com/MetaMask/metamask-extension/pull/15383))
 - [FLASK] Fix key management permission copy ([#14849](https://github.com/MetaMask/metamask-extension/pull/14849))
```
Comment (FrederikBolding) [inline CHANGELOG.md:73]:
```suggestion
- [FLASK] Fix Snaps key management permission copy ([#14849](https://github.com/MetaMask/metamask-extension/pull/14849))
```
Can be categorized as "fixed"

File: CHANGELOG.md [hunk 4]
```diff
@@ -6,232 +6,71 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
-- Bump @metamask/providers from 9.0.0 to 9.1.0 ([#15514](https://github.com/MetaMask/metamask-extension/pull/15514))
-- Fix onboarding events ([#15608](https://github.com/MetaMask/metamask-extension/pull/15608))
-- Add test case to ensure that action addCustomNetwork is idempotent ([#15602](https://github.com/MetaMask/metamask-extension/pull/15602))
-- added avatar network component ([#15502](https://github.com/MetaMask/metamask-extension/pull/15502))
-- filter breadcrumbs for improved clarity while debugging sentry errors ([#15639](https://github.com/MetaMask/metamask-extension/pull/15639))
-- Fix state creation in setupSentryGetStateGlobal ([#15635](https://github.com/MetaMask/metamask-extension/pull/15635))
-- UI-14395: typography v2 ([#15408](https://github.com/MetaMask/metamask-extension/pull/15408))
-- added avatar token component ([#15466](https://github.com/MetaMask/metamask-extension/pull/15466))
-- Fix Signature Request styles ([#15621](https://github.com/MetaMask/metamask-extension/pull/15621))
 - [FLASK] `snaps-skunkworks@0.19.0` ([#15619](https://github.com/MetaMask/metamask-extension/pull/15619))
```
Comment (FrederikBolding) [inline CHANGELOG.md:71]:
```suggestion
- [FLASK] Add `snap_getBip32Entropy` JSON-RPC method ([#15619](https://github.com/MetaMask/metamask-extension/pull/15619))
```
Can be categorized as "added"

File: CHANGELOG.md (hunk 1 above)
Comment (BeroBurny) [reply CHANGELOG.md:51]: thanks for the fast response, now need to prepare PRs for upcoming changes

File: CHANGELOG.md [hunk 5]
```diff
@@ -6,232 +6,71 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Update title case to sentence case with some exceptions (names, companies, special terms and page navigation) ([#15285](https://github.com/MetaMask/metamask-extension/pull/15285))
+- Swaps: show a block explorer URL and hide "Add token" prompt, if block explorer URL is not available on Build Quote screen ([#15198](https://github.com/MetaMask/metamask-extension/pull/15198))
+- Swaps: Standardize the spelling of 'cancelled' ([#15266](https://github.com/MetaMask/metamask-extension/pull/15266))
+- Swaps: Performance improvements for Swaps, especially on the Build Quote page (~7.2x faster) ([#15359](https://github.com/MetaMask/metamask-extension/pull/15359))
+- Swaps: only check if a user has enough balance before calling STX. If "Not enough funds" error is thrown, show a notification, but keep STX active ([#15218](https://github.com/MetaMask/metamask-extension/pull/15218))
+- [FLASK] Disable "Mark all as read button" when there are no notifications ([#15333](https://github.com/MetaMask/metamask-extension/pull/15333))
+- [FLASK] Update Snaps packages to version `^0.18.1` ([#15230](https://github.com/MetaMask/metamask-extension/pull/15230))
```
Comment (FrederikBolding) [inline CHANGELOG.md:45]: I think this one can be omitted. Not much exciting user-facing happened in this update.

File: CHANGELOG.md [hunk 6]
```diff
@@ -6,232 +6,71 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
-- Updated Avalanche rpcUrl endpoint ([#15720](https://github.com/MetaMask/metamask-extension/pull/15720))
-- Updated the main copy on SetApprovalForAll confirmation screen ([#15724](https://github.com/MetaMask/metamask-extension/pull/15724))
-- Set view full transaction details by default ([#15726](https://github.com/MetaMask/metamask-extension/pull/15726))
-- Make updateTransactionSendFlowHistory background method idempotent ([#15585](https://github.com/MetaMask/metamask-extension/pull/15585))
-- Make addUnapprovedTransaction action idempotent ([#15667](https://github.com/MetaMask/metamask-extension/pull/15667))
-- improvement it translation ([#15477](https://github.com/MetaMask/metamask-extension/pull/15477))
-- MV3: add retry logic to actions ([#15337](https://github.com/MetaMask/metamask-extension/pull/15337))
-- Unit tests for first time flow/new account and onboarding flow. ([#15625](https://github.com/MetaMask/metamask-extension/pull/15625))
-- Set lavamoat codeowners to allow supply chain reviews ([#15624](https://github.com/MetaMask/metamask-extension/pull/15624))
 - [FLASK] `snaps-skunkworks@0.20.0` ([#15706](https://github.com/MetaMask/metamask-extension/pull/15706))
```
Comment (FrederikBolding) [inline CHANGELOG.md:70]:
```suggestion
- [FLASK] Add `snap_getBip44Entropy` RPC method and deprecate `snap_getBip44Entropy_*` ([#15706](https://github.com/MetaMask/metamask-extension/pull/15706))
```
Can be categorized as "added".

File: CHANGELOG.md (hunk 5 above)
Comment (seaona) [reply CHANGELOG.md:45]: @FrederikBolding do you confirm is it safe to remove the complete entry and sub-entries?

```
- [FLASK] Update Snaps packages to version `^0.18.1` ([#15230](https://github.com/MetaMask/metamask-extension/pull/15230))
  - Updates the following packages from `0.16.0` to `0.18.1`:
    - `@metamask/execution-environments`
    - `@metamask/rpc-methods`
    - `@metamask/snap-controllers`
    - `@metamask/snap-types`
  - Serialization issues were fixed in `@metmask/execution-environments`. See the ([releases](https://github.com/MetaMask/snaps-skunkworks/releases)) of the Snaps monorepo for details.
```

Comment (FrederikBolding) [reply CHANGELOG.md:48]: Yeah. No real user-facing changes were introduced in this change. I also suggested that we change the other bump PRs to list the main user-facing features instead.

We could include every time we update the Snaps packages, but it doesn't seem that useful imo.

Comment (seaona) [reply CHANGELOG.md]: Thanks @FrederikBolding I've removed the item and sub-items. Your suggestion sounds something we could talk on one of our Release meetings on Thursday. Feel free to jump in any week and we could further discuss it, if that sounds okay to you :)

**Review by brad-decker — APPROVED** [review]

**Pull request merged**